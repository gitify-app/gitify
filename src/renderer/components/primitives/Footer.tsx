import { Box, Stack } from '@primer/react';
import type { FC, ReactNode } from 'react';

interface IFooter {
  children: ReactNode;
  justify: 'end' | 'space-between';
}

export const Footer: FC<IFooter> = (props: IFooter) => {
  return (
    <Box className="fixed bottom-0 left-10 right-0 bg-gitify-footer">
      <Stack direction="horizontal" justify={props.justify} padding="condensed">
        {props.children}
      </Stack>
    </Box>
  );
};
