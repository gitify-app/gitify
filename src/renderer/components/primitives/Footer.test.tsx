import { render } from '@testing-library/react';
import { Footer } from './Footer';

describe('renderer/components/primitives/Footer.tsx', () => {
  it('should render itself & its children - justify-between', () => {
    const tree = render(<Footer justify="justify-between">Test</Footer>);

    expect(tree).toMatchSnapshot();
  });

  it('should render itself & its children - justify-end', () => {
    const tree = render(<Footer justify="justify-end">Test</Footer>);

    expect(tree).toMatchSnapshot();
  });
});
