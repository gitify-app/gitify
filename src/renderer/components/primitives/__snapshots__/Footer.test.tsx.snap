// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer/components/primitives/Footer.tsx should render itself & its children - end 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Box-sc-g0xbh4-0 fixed bottom-0 left-10 right-0 bg-gitify-footer"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
          data-align="stretch"
          data-direction="horizontal"
          data-justify="end"
          data-padding="condensed"
          data-wrap="nowrap"
        >
          Test
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-sc-g0xbh4-0 fixed bottom-0 left-10 right-0 bg-gitify-footer"
    >
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
        data-align="stretch"
        data-direction="horizontal"
        data-justify="end"
        data-padding="condensed"
        data-wrap="nowrap"
      >
        Test
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renderer/components/primitives/Footer.tsx should render itself & its children - space-between 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Box-sc-g0xbh4-0 fixed bottom-0 left-10 right-0 bg-gitify-footer"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
          data-align="stretch"
          data-direction="horizontal"
          data-justify="space-between"
          data-padding="condensed"
          data-wrap="nowrap"
        >
          Test
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-sc-g0xbh4-0 fixed bottom-0 left-10 right-0 bg-gitify-footer"
    >
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
        data-align="stretch"
        data-direction="horizontal"
        data-justify="space-between"
        data-padding="condensed"
        data-wrap="nowrap"
      >
        Test
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
